keyboards/ergodox_ez/keymaps/hacker_dvorak/keymap.c

```diff
@@ -0,0 +1,797 @@
+                        XXXXXXX,
+        RESET, XXXXXXX, XXXXXXX,
+
+        // right hand
+        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, MS_GE,
+        MN_GUI,  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, MU_GE,
+                 XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, MT_NKRO,
+        MY_GUI,  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, MT_AG,
+                          XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+
+        // right thumb
+        XXXXXXX, XXXXXXX,
+        XXXXXXX,
+        XXXXXXX, MS_CC,   MU_CC
+    )
+};
+
+const uint16_t PROGMEM fn_actions[] = {
+    [1] = ACTION_LAYER_TAP_TOGGLE(1)
+};
+
+void matrix_init_user(void) {
+    rgblight_sethsv_noeeprom_green();
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        // dynamically generate these.
+        case EPRM:
+            if (record->event.pressed) {
+                eeconfig_init();
+            }
+
+            return false;
+
+        case RGB_SLD:
+            if (record->event.pressed) {
+                rgblight_mode(1);
+            }
+
+            return false;
+
+        case RGB_RED:
+            if (record->event.pressed) {
+                #ifdef RGBLIGHT_ENABLE
+                    rgblight_enable();
+                    rgblight_mode(1);
+                    rgblight_sethsv_noeeprom_red();
+                #endif
+            }
+
+            return false;
+
+        case RGB_GREEN:
+            if (record->event.pressed) {
+                #ifdef RGBLIGHT_ENABLE
+                    rgblight_enable();
+                    rgblight_mode(1);
+                    rgblight_sethsv_noeeprom_green();
+                #endif
+            }
+
+            return false;
+
+        case RGB_BLUE:
+            if (record->event.pressed) {
+                #ifdef RGBLIGHT_ENABLE
+                    rgblight_enable();
+                    rgblight_mode(1);
+                    rgblight_sethsv_noeeprom_blue();
+                #endif
+            }
+
+            return false;
+
+        case RGB_MAGENTA:
+            if (record->event.pressed) {
+                #ifdef RGBLIGHT_ENABLE
+                    rgblight_enable();
+                    rgblight_mode(1);
+                    rgblight_sethsv_noeeprom_magenta();
+                #endif
+            }
+
+            return false;
+
+        case RGB_YELLOW:
+            if (record->event.pressed) {
+                #ifdef RGBLIGHT_ENABLE
+                    rgblight_enable();
+                    rgblight_mode(1);
+                    rgblight_sethsv_noeeprom_yellow();
+                #endif
+        }
+
+        return false;
+
+        default:
+            return true;
+    }
+}
+
+uint32_t layer_state_set_user(uint32_t state) {
+    uint8_t layer = biton32(state);
+
+    ergodox_board_led_off();
+    ergodox_right_led_1_off();
+    ergodox_right_led_2_off();
+    ergodox_right_led_3_off();
+
+    switch (layer) {
```

Comment:
You should be fine with calling it directly, like:
```suggestion
    switch (biton32(layer)) {
```